### ISSUE TITLE
Cannot set EMSCRIPTEN_ROOT to a softlink

### ISSUE THREAD

**Issue by fduxiao** [kripken/emscripten#6929]
I use homebrew to install emscripten, which puts emscripten root at `/usr/local/Cellar/emscripten/1.38.10/libexec`. 
While I try to set it to `/usr/local/opt/emscripten/libexec`(so I can get rid of the hard-coded version), emcc always complaints that `ERROR:root:Incorrect EMSCRIPTEN_ROOT in config file: /usr/local/opt/emscripten/libexec (Expected /usr/local/Cellar/emscripten/1.38.10/libexec)`.
Everything goes just fine if I set it to  `/usr/local/Cellar/emscripten/1.38.10/libexec`.
Is there any way to config without the versioned path? 

My environment:
```shell
$ uname -a
Darwin somehostname 17.6.0 Darwin Kernel Version 17.6.0: Tue May  8 15:22:16 PDT 2018; root:xnu-4570.61.1~1/RELEASE_X86_64 x86_64
```
```shell
$ brew --version 
Homebrew 1.7.1
Homebrew/homebrew-core (git revision b9ed4; last commit 2018-08-01)
```
```shell
$ emcc --version
emcc (Emscripten gcc/clang-like replacement) 1.38.10 ((unknown revision))
Copyright (C) 2014 the Emscripten authors (see AUTHORS.txt)
This is free and open source software under the MIT license.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
```shell
$ realpath /usr/local/Cellar/emscripten/1.38.10/libexec
/usr/local/Cellar/emscripten/1.38.10/libexec
```
```shell
$ realpath /usr/local/opt/emscripten/libexec
/usr/local/Cellar/emscripten/1.38.10/libexec
```
```shell
$ readlink /usr/local/opt/emscripten
../Cellar/emscripten/1.38.10
```
```shell
$ head $HOME/.emscripten  # other lines are generated automatically
```
```python
# Note: If you put paths relative to the home directory, do not forget os.path.expanduser

# Note: On Windows, remember to escape backslashes! I.e. EMSCRIPTEN_ROOT='c:\emscripten\' is not valid, but EMSCRIPTEN_ROOT='c:\\emscripten\\' and EMSCRIPTEN_ROOT='c:/emscripten/' are.

import os

# this helps projects using emscripten find it
EMSCRIPTEN_ROOT = os.path.expanduser(os.getenv('EMSCRIPTEN', '/usr/local/opt/emscripten/libexec')) # directory
LLVM_ROOT = os.path.expanduser(os.getenv('LLVM', '/usr/local/opt/emscripten/libexec/llvm/bin')) # directory
```
```shell
$ which emcc  
/usr/local/bin/emcc
```
```shell
$ cat > /tmp/tmp.c << EOF 
#include <stdio.h>

int main()
{
    printf("hello world\n");
    return 0;
}

EOF
```
```shell
$ cc /tmp/tmp.c -o /tmp/a.out
$ /tmp/a.out
hello world
```
```shell
$ emcc /tmp/tmp.c
ERROR:root:Incorrect EMSCRIPTEN_ROOT in config file: /usr/local/opt/emscripten/libexec (Expected /usr/local/Cellar/emscripten/1.38.10/libexec)
```
```shell
$ head $HOME/.emscripten 
```
```python

# Note: If you put paths relative to the home directory, do not forget os.path.expanduser

# Note: On Windows, remember to escape backslashes! I.e. EMSCRIPTEN_ROOT='c:\emscripten\' is not valid, but EMSCRIPTEN_ROOT='c:\\emscripten\\' and EMSCRIPTEN_ROOT='c:/emscripten/' are.

import os

# this helps projects using emscripten find it
EMSCRIPTEN_ROOT = os.path.expanduser(os.getenv('EMSCRIPTEN', '/usr/local/Cellar/emscripten/1.38.10/libexec')) # directory
LLVM_ROOT = os.path.expanduser(os.getenv('LLVM', '/usr/local/opt/emscripten/libexec/llvm/bin')) # directory
```
```shell
$ emcc /tmp/tmp.c -o /tmp/a.out.js
INFO:root:generating system asset: is_vanilla.txt... (this will be cached in "/Users/xiao/.emscripten_cache/is_vanilla.txt" for subsequent builds)
INFO:root: - ok
WARNING:root:(Emscripten: settings file has changed, clearing cache)
INFO:root:(Emscripten: Running sanity checks)
INFO:root:generating port: binaryen_tag_version_49.txt... (this will be cached in "/Users/xiao/.emscripten_cache/asmjs/binaryen_tag_version_49.txt" for subsequent builds)
INFO:root:building port: binaryen
-- Building with -std=c++11
-- Building with -Wall
-- Building with -Werror
-- Building with -Wextra
-- Building with -Wno-unused-parameter
-- Building with -fno-omit-frame-pointer
-- Building with -fPIC
-- Building with -O2
-- Building with -UNDEBUG
-- Configuring done
CMake Warning (dev):
  Policy CMP0042 is not set: MACOSX_RPATH is enabled by default.  Run "cmake
  --help-policy CMP0042" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  MACOSX_RPATH is not specified for the following targets:

   binaryen

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /Users/xiao/.emscripten_ports/binaryen/binaryen-version_49
[  1%] Linking CXX static library ../../lib/libasmjs.a
[  2%] Linking CXX static library ../../lib/libir.a
[  2%] Linking CXX static library ../../lib/libcfg.a
[  3%] Linking CXX static library ../../lib/libwasm.a
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: ../../lib/libwasm.a(wasm-interpreter.cpp.o) has no symbols
[  5%] Built target cfg
[  8%] Built target asmjs
[ 10%] Built target ir
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: ../../lib/libwasm.a(wasm-interpreter.cpp.o) has no symbols
[ 11%] Linking CXX static library ../../lib/libemscripten-optimizer.a
[ 20%] Built target wasm
[ 21%] Linking CXX static library ../../lib/libsupport.a
[ 68%] Built target passes
[ 76%] Built target emscripten-optimizer
[ 78%] Built target support
[ 79%] Linking CXX executable bin/wasm-ctor-eval
[ 81%] Linking CXX executable bin/wasm-emscripten-finalize
[ 82%] Linking CXX executable bin/wasm-reduce
[ 82%] Linking CXX shared library lib/libbinaryen.dylib
[ 83%] Built target wasm-emscripten-finalize
[ 85%] Built target wasm-reduce
[ 85%] Built target wasm-ctor-eval
[ 85%] Built target binaryen
[ 86%] Linking CXX executable bin/wasm2asm
[ 87%] Linking CXX executable bin/wasm-metadce
[ 88%] Linking CXX executable bin/asm2wasm
[ 89%] Linking CXX executable bin/wasm-shell
[ 90%] Built target wasm-metadce
[ 91%] Built target wasm-shell
[ 92%] Built target wasm2asm
[ 92%] Built target asm2wasm
[ 93%] Linking CXX executable bin/wasm-dis
[ 94%] Linking CXX executable bin/wasm-opt
[ 95%] Linking CXX executable bin/wasm-as
[ 96%] Linking CXX executable bin/wasm-merge
[ 97%] Built target wasm-dis
[ 98%] Built target wasm-as
[ 99%] Built target wasm-opt
[100%] Built target wasm-merge
INFO:root: - ok
INFO:root:generating system library: dlmalloc.bc... (this will be cached in "/Users/xiao/.emscripten_cache/asmjs/dlmalloc.bc" for subsequent builds)
INFO:root: - ok
INFO:root:generating system library: libc.bc... (this will be cached in "/Users/xiao/.emscripten_cache/asmjs/libc.bc" for subsequent builds)
INFO:root: - ok
INFO:root:generating system library: wasm-libc.bc... (this will be cached in "/Users/xiao/.emscripten_cache/asmjs/wasm-libc.bc" for subsequent builds)
INFO:root: - ok
INFO:root:generating system asset: generated_struct_info.json... (this will be cached in "/Users/xiao/.emscripten_cache/asmjs/generated_struct_info.json" for subsequent builds)
INFO:root: - ok

```
```shell
$ node /tmp/a.out.js
hello world
```

**Comment by kripken** [comment]
This should be fixed on latest incoming (we removed the assert). No new tagged version with it yet, but you can just delete the assertion manually, or use incoming for now.

**Comment by fduxiao** [comment]
It has been fixed in version `1.38.11` :) .
